test/ng/directive/ngOptionsSpec.js

```diff
@@ -1657,7 +1701,7 @@ describe('ngOptions', function() {
         scope.values.pop();
       });
 
-      expect(element.val()).toEqual('');
+      expect(element.val()).toEqualUnknownValue();
```

Comment:
OK, so the reason for this is that, if the model is `null` then `element.val()` is empty but if the model contains a value that doesn't match any option then `element.val()` is `'?'`.

When we remove the option the value becomes invalid so we get `'?'` but then the model is cleared out and becomes `null`. If we have a watch on the model then this will cause the `ngModel` to run `$render`, which update the element value to `''`.

I think that what we need is to trigger a new call to `$render` when the selected option is removed...